### ISSUE TITLE
Cannot restore Shield from backups

### ISSUE THREAD

**Issue by Bulbuc** [shieldproject/shield#443]
Hello,

I deployed a new Shield and tried to restore it from the backup archive, but without success.
I even re-keyed Shield, made another backup, and selected this key+file combination , but the same result.
As a side note, source shield version is 8.0.13 and target recovery VM has 8.0.14

Shield job:
```
UUID                                  Name   Summary  Schedule       Status  Policy              SHIELD Agent        Target                    Store            Fixed-Key
====                                  ====   =======  ========       ======  ======              ============        ======                    =====            =========
3d99679a-87de-47a3-85d8-cb04899c1e2b  Daily           daily at 6:00          Short-term-2-week   127.0.0.1:5444      Shield - itself           Openstack Swift  true
```

Shield target:
```
UUID                                  Name                      Summary                   Plugin      SHIELD Agent        Configuration
====                                  ====                      =======                   ======      ============        =============
db01bac8-ae2d-402f-bd72-f439573c4c5c  Shield - itself           Shield - itself           fs          127.0.0.1:5444      {
                                                                                                                           "base_dir": "/var/vcap/store/shield",
                                                                                                                           "exclude": "*.db-journal"
                                                                                                                          }
```

```
SHIELD Self-restore started on Wed, 08 Aug 2018 11:08:38 UTC
Validating environment...
=========================
SHIELD_OP ... found
SHIELD_STORE_PLUGIN ... found
SHIELD_STORE_ENDPOINT ... found
SHIELD_TARGET_PLUGIN ... found
SHIELD_TARGET_ENDPOINT ... found
SHIELD_RESTORE_KEY ... found
OK

Validating TARGET plugin `fs`...
================================
✓ base_dir  files in /var/vcap/store will be backed up
✓ include   all files will be included
✓ exclude   no files will be excluded
OK

Validating STORE plugin `fs`...
===============================
✓ base_dir  files in /tmp/ will be backed up
✓ include   all files will be included
✓ exclude   no files will be excluded
OK

Running restore task (using bzip2 compression)
==============================================
panic: cipher.NewCTR: IV length must equal block size

goroutine 1 [running]:
crypto/cipher.NewCTR(0x7df440, 0xc42006ade0, 0xc4200f4000, 0xf0, 0xf0, 0x0, 0x0)
	/goroot/src/crypto/cipher/ctr.go:38 +0x322
github.com/starkandwayne/shield/crypter.Stream(0xc4200144d0, 0xa, 0xc4200144e0, 0x10, 0x10, 0xc4200f4000, 0xf0, 0xf0, 0x1e0, 0xc42004bf50, ...)
	/tmp/build/01f561ac/gopath/src/github.com/starkandwayne/shield/crypter/crypter.go:49 +0x426
main.main()
	/tmp/build/01f561ac/gopath/src/github.com/starkandwayne/shield/cmd/shield-crypt/main.go:40 +0x4fb

bunzip2: Compressed file ends unexpectedly;
	perhaps it is corrupted?  *Possible* reason follows.
bunzip2: Inappropriate ioctl for device
	Input file = (stdin), output file = (stdout)

It is possible that the compressed file(s) have become corrupted.
You can use the -tvv option to test integrity of such files.

You can use the `bzip2recover' program to attempt to recover
data from undamaged sections of corrupted files.


EXITING 2
```

**Comment by jhunt** [comment]
We made some progress on this yesterday; the crux of the issue is that for out-of-band init-restore scenarios, the Fixed Key was not being properly converted into a key+iv for decrypting the backup.  We fixed that, and the panics go away, but we found some anomalies with the restored SHIELD that we would like to investigate before releasing a patch.

**Comment by jhunt** [comment]
In the process of rewriting the task scheduler (which the self-restore was an expertly ugly set of edge cases), I believe we found some mismatch between the handling of the fixed key material and the derivation of key+iv for encryption/decryption.

This has been fixed and vigorously tested in v8.1., due out next week at CF Summit Philadelphia.  Existing Fixed Keys and their associated archives should be viable after the fixes in 8.1. are deployed.

**Comment by jhunt** [comment]
@patjones and I worked through testing this, rigorously, in our lab, and noticed a few issues related to pathing that have been fixed in #525.  We will be documenting this process, and the internal details of how it works, in a forthcoming blog post, in advance of an 8.2 release.  I'll drop a note here when that release happens, so you can try that, @Bulbuc

**Comment by jhunt** [comment]
Whelp, this one fell off my radar.  @Bulbuc have you per chance seen and/or tried to deploy the 8.2.1 SHIELD to see if this fixes your issues?
